server/repositories/entities/field.ts

```diff
@@ -0,0 +1,22 @@
+import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm'
+
+@Entity('field')
+export default class Field {
+  @PrimaryGeneratedColumn()
+  id: number
+
+  @Column()
+  type: string
+
+  @Column()
+  name: string
+
+  @Column()
+  required: boolean
+
+  @Column({ nullable: true })
+  validation: string
+
+  @OneToMany('Field', 'id')
+  field: Array<Field> | null
```

Comment:
Ought this to be `fields`? Also is null valid here? Would an empty array not be better?

Reply:
Aha, that's not correct and should be a OneToOne. Updated accordingly.